packages/backend-common/src/util/docker.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Docker from 'dockerode';
+import { PassThrough, Writable } from 'stream';
+
+export type UserOptions = {
+  User?: string;
+};
+
+export type RunDockerContainerOptions = {
+  imageName: string;
+  args: string[];
+  logStream?: Writable;
+  dockerClient: Docker;
+  mountDirs?: Map<string, string>;
+  workingDir?: string;
+  envVars?: string[];
```

Comment:
Kinda thinking it would be nice to pass pass this as `env?: Record<string, string>` too. Node.js usually uses objects to represent the env. Not super opinionated about this though